subprojects/docs/src/docs/userguide/authoring-builds/introduction/plugins.adoc

```diff
@@ -0,0 +1,579 @@
+
+*Binary plugins* are written either programmatically by implementing the link:{javadocPath}/org/gradle/api/Plugin.html[`Plugin`] interface or through declarations in either Groovy or Kotlin DSL.
+They can reside within a build script, the project hierarchy, or externally in a plugin jar.
+
+*Script plugins* are additional build scripts that further configure the build and usually implement a declarative approach to manipulating the build.
+They are typically used within a build but can be externalized and accessed remotely.
+
+A plugin often starts as a script plugin (because they are easy to write).
+Then, as the code becomes more valuable, it's migrated to a binary plugin that can be easily tested and shared between multiple projects or organizations.
+
+[[sec:using_plugins]]
+== Using plugins
+
+To use the build logic encapsulated in a plugin, Gradle needs to perform two steps.
+First, it needs to *resolve* the plugin, and then it needs to *apply* the plugin to the target, usually a link:{groovyDslPath}/org.gradle.api.Project.html[`Project`].
+
+*Resolving* a plugin means finding the correct version of the jar that contains a given plugin and adding it to the script classpath.
+Once a plugin is resolved, its API can be used in a build script.
+Script plugins are self-resolving in that they are resolved from the specific file path or URL provided when applying them.
+Core binary plugins provided as part of the Gradle distribution are automatically resolved.
+
+*Applying* a plugin means executing the plugin's link:{javadocPath}/org/gradle/api/Plugin.html#apply-T-[Plugin.apply(T)] on the Project you want to enhance with the plugin.
+Applying plugins is _idempotent_.
+You can safely apply any plugin multiple times without side effects.
+
+The <<#sec:plugins_block,plugins DSL>> is recommended to resolve and apply plugins in one step.
+
+[[sec:binary_plugin_locations]]
+== Resolving plugins
+
+Gradle provides the *core plugins* (e.g., `JavaPlugin`, `GroovyPlugin`, `MavenPublishPlugin`, etc.) as part of its distribution, which means they are automatically resolved.
+
+However, non-core plugins must be resolved before they can be applied.
+This can be achieved in several ways:
+
+[%autowidth.stretch]
+|===
+|# |Application |Recommended Usage |Where
+
+|<<#sec:plugins_block,1>>
+|<<#sec:plugins_block,Using the `plugins` block>>
+|Applying *community* plugin or *local* plugins in `buildSrc` to a specific project.
+|Build script or Settings file
+
+||
+a|
+[source,kotlin]
+----
+plugins {
+  id("org.barfuin.gradle.taskinfo") version "2.1.0"
+}
+----
+|
+
+|<<#sec:buildsrc_plugins_dsl,2>>
+|<<#sec:buildsrc_plugins_dsl,Using the `buildSrc` directory>>
+|Applying *community* or *local* plugins to multiple subprojects.
+|Build script
+
+||
+a|
+[source,kotlin]
+----
+plugins {
+    id("org.barfuin.gradle.taskinfo") version "2.1.0"
+}
+repositories {
+    jcenter()
+}
+dependencies {
+    implementation(Libs.Kotlin.coroutines)
```

Comment:
❗ What is `Libs.Kotlin` here?  Why introduce VCs for just this example at this point?